pages/api/id/[form]/retrieval.ts

```diff
@@ -0,0 +1,207 @@
+  if (result) {
+    const { formResponses } = result as ResponseListResult;
+    // log level is warn because in production level info is not being forwarded to CloudWatch
+    logMessage.warn(
+      `user:${email} retrieved form responses [${formResponses.Items?.map(
+        (response) => response.SubmissionID
+      )}] from form ID:${formID} at:${Date.now()} using token:${temporaryToken}`
+    );
+
+    return res.status(200).json({ responses: formResponses.Items });
+  }
+}
+
+/**
+ * Request type: DELETE
+ * curl -X DELETE http://gc-forms/api/id/1/retrieval
+     -H "Content-Type: application/json"
+     -d '["submissionID1", "submissionID2"]'
+ * This method will delete the specified form responses
+ * @param req - The NextJS request object
+ * @param res - The NextJS response object
+ * @param formID - The form ID from which to retrieve responses
+ * @param email - The email of the authenticated user carrying out the request
+ * @param temporaryToken - The valid JWT temporary token of the authenticated user carrying out the request
+ */
+async function deleteFormResponses(
+  req: NextApiRequest,
+  res: NextApiResponse,
+  formID: string,
+  email?: string,
+  temporaryToken?: string
+) {
+  // get body from request
+  const requestBody = req.body as string[];
+  const submissionIDlist: string[] = [];
+  try {
+    const documentClient = connectToDynamo();
+    //Setting Retrieved to 1. DynamoDB stream will trigger archival lambda
+    for (const submissionID of requestBody) {
+      const updateItem = {
+        TableName: "Vault",
+        Key: {
+          SubmissionID: submissionID,
+          FormID: formID,
+        },
+        UpdateExpression: "SET Retrieved = :retrieved",
+        ConditionExpression: "SubmissionID = :submission",
+        ExpressionAttributeValues: {
+          ":retrieved": 1,
+          ":submission": submissionID,
+        },
+        ReturnValues: "NONE",
+      };
+      //Update one item at the time
+      await documentClient.send(new UpdateCommand(updateItem));
+      submissionIDlist.push(submissionID);
+    }
+
+    // log level is warn because in production level info is not being forwarded to CloudWatch
+    logMessage.warn(
+      `user:${email} marked form responses [${submissionIDlist}] from form ID:${formID} as retrieved at:${Date.now()} using token:${temporaryToken}`
+    );
+
+    //Return responses data
+    return res.status(200).json(submissionIDlist);
+  } catch (error) {
+    logMessage.warn("Some submissions were potentially not marked as retrieved");
+    logMessage.warn(
+      submissionIDlist
+        ? `The following submissions were not marked as retrieved: [${requestBody.filter(
+            (submissionID) => !submissionIDlist.includes(submissionID)
+          )}]`
+        : `[${requestBody.toString()}]`
+    );
+    logMessage.error(error as Error);
+    return res.status(500).json({ error: "Error on Server Side" });
+  }
+}
+
+// Only a GET request is allowed
```

Comment:
Not up to date but thinking we could get rid of this since the middleware function is pretty transparent on what is allowed ;)

Reply:
Yup I agree thanks for the catch